bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java

```diff
@@ -593,24 +596,49 @@ private void applyDefaultConfiguration(Thing thing, ThingType thingType) {
     }
 
     private boolean isInitializable(Thing thing, ThingType thingType) {
-        // determines if all 'required' configuration parameters are available in the configuration
-        if (thingType == null) {
-            logger.debug("Thing type for thing {} is not known, assuming it is initializable", thing.getUID());
+        if (!isComplete(thingType, thing.getUID(), tt -> tt.getConfigDescriptionURI(), thing.getConfiguration())) {
+            return false;
+        }
+
+        for (Channel channel : thing.getChannels()) {
+            ChannelType channelType = channelTypeRegistry.getChannelType(channel.getChannelTypeUID());
+            if (!isComplete(channelType, channel.getUID(), ct -> ct.getConfigDescriptionURI(),
+                    channel.getConfiguration())) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Determines if all 'required' configuration parameters are available in the configuration
+     *
+     * @param prototype
+     * @param targetUID
+     * @param configDescriptionURIFunction
+     * @param configuration
+     * @return
+     */
+    private <T extends Identifiable<?>> boolean isComplete(T prototype, UID targetUID,
+            Function<T, URI> configDescriptionURIFunction, Configuration configuration) {
+        if (prototype == null) {
+            logger.debug("Prototype for '{}' is not known, assuming it is initializable", targetUID);
             return true;
         }
 
-        ConfigDescription description = resolve(thingType.getConfigDescriptionURI(), null);
+        ConfigDescription description = resolve(configDescriptionURIFunction.apply(prototype), null);
         if (description == null) {
-            logger.debug("Config description for thingtype {} is not resolvable, assuming thing {} is initializable",
-                    thingType.getUID(), thing.getUID());
+            logger.debug("Config description for '{}' is not resolvable, assuming '{}' is initializable",
+                    prototype.getUID(), targetUID);
             return true;
         }
 
         List<String> requiredParameters = getRequiredParameters(description);
-        Map<String, Object> properties = thing.getConfiguration().getProperties();
+        Map<String, Object> properties = configuration.getProperties();
         if (logger.isDebugEnabled()) {
-            logger.debug("Configuration of thing {} needs {}, has {}.", thing.getUID(), requiredParameters,
-                    thing.getConfiguration().getProperties().keySet());
+            logger.debug("Configuration of '{}' needs {}, has {}.", targetUID, requiredParameters,
+                    configuration.getProperties().keySet());
```

Comment:
As you already store `configuration.getProperties()` to `properties`, no need to call `getProperties()` again.